packages/components/inputs/select-utils/src/clear-indicator/clear-indicator.tsx

```diff
@@ -1,11 +1,21 @@
-import PropTypes from 'prop-types';
+import type { CSSProperties, LegacyRef } from 'react';
+import type { ClearIndicatorProps } from 'react-select';
 import { css } from '@emotion/react';
 import { useIntl } from 'react-intl';
 import { customProperties as vars } from '@commercetools-uikit/design-system';
 import { CloseIcon } from '@commercetools-uikit/icons';
 import messages from './messages';
 
-const ClearIndicator = (props) => {
+type TInnerProps = {
+  ref: LegacyRef<HTMLButtonElement>;
+} & JSX.IntrinsicElements['button'];
+
+type TClearIndicator = {
+  getStyles: (s: string, props: unknown) => CSSProperties;
+  innerProps: TInnerProps;
+} & ClearIndicatorProps;
+
+const ClearIndicator = (props: TClearIndicator) => {
```

Comment:
According to this example, we should be able to use `ClearIndicatorProps` directly. Can you double check?

https://github.com/JedWatson/react-select/blob/4e06d22e47bbd82a4f9f67452c7f16787c4ce529/docs/examples/CustomClearIndicator.tsx#L8-L13

Reply:
Yes, we can use it directly. However, we need to update the ref type to use HTMLButtonElement since we are using a button instead of a div.
The `getStyles`  type def has been cleaned. 
Thanks.